scalyr_agent/agent_main.py

```diff
@@ -690,7 +699,10 @@ def __detailed_status(self, data_directory, status_format="text"):
             fp.write(status_format)
 
         # Signal to the running process.  This should cause that process to write to the status file
-        result = self.__controller.request_agent_status()
+        if health_check:
+            result = self.__controller.request_agent_health_check()
```

Comment:
This also should allow you to remove a bunch of the changes you made related to registering and handling the new health check handler.